util/queue.go

```diff
@@ -0,0 +1,58 @@
+package util
+
+import (
+	"errors"
+	"sync"
+)
+
+type Queue struct {
```

Comment:
Its my preference to avoid writing "generic" implementations (which are really just type-unsafe implemenations) of datastructures/interfaces like this, and just have things like DurationQueue/etc (which we could codgen eventually.

Reply:
made this specific + private